### PR TITLE
feat(types): provide augmentations for `only` and `without`

### PR DESCRIPTION
**Pull request by harlan-zw**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Allows type hinting on the keys for `only` and `without` as well as augmenting the result which will either include or exclude the provided keys from the IntelliSense respectively.

image

image

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Atinux — APPROVED** [review]
Looks good to me! Awesome change. Maybe @farnabaz can have a second review

**Comment by Tahul** [comment]
LGTM :)

Thanks! ❤️

File: src/runtime/query/query.ts
Comment (farnabaz) [inline src/runtime/query/query.ts:7]:
```suggestion
export const createQuery = <T = ParsedContent>(
```

File: src/runtime/query/query.ts
Comment (farnabaz) [inline src/runtime/query/query.ts:2]:
```suggestion
import { ParsedContent } from '../types'
```

**Review by farnabaz — APPROVED** [review]

**Review by Tahul — APPROVED** [review]

**Pull request merged**
